service/src/test/java/org/broadinstitute/listener/relay/http/RelayedHttpRequestProcessorTest.java

```diff
@@ -178,18 +179,63 @@ void writeTargetResponseOnCaller_responseIsWrittenBackToCaller() throws IOExcept
 
   @Test
   void writeTargetResponseOnCaller_removesInvalidAzureRelayHeaders() throws IOException {
+    validateHeaderRemoval("Transfer-Encoding", "chunked");
+    validateHeaderRemoval("transfer-encoding", "chunked");
+  }
+
+  @Test
+  /** @see https://broadworkbench.atlassian.net/browse/IA-4478 */
+  void writeTargetResponseOnCaller_removesServerHeaders() throws IOException {
+    validateHeaderRemoval("Server", "Microsoft-HTTPAPI/2.0");
+  }
+
+  private void validateHeaderRemoval(String headerKey, String headerVal) {
+    Map<String, String> listenerHeaders = mock(HashMap.class);
+    when(listenerResponse.getHeaders()).thenReturn(listenerHeaders);
+
     when(targetHttpResponse.getContext()).thenReturn(context);
-    when(targetHttpResponse.getBody()).thenReturn(Optional.of(body));
-    when(targetHttpResponse.getStatusCode()).thenReturn(200);
-    Map<String, String> headers = new HashMap();
-    headers.put("transfer-encoding", "chunked");
-    when(targetHttpResponse.getHeaders()).thenReturn(Optional.of(headers));
     when(context.getResponse()).thenReturn(listenerResponse);
+
+    when(targetHttpResponse.getStatusCode()).thenReturn(200);
+
+    // skip #getStatusDescription(...) validation
+
+    Map<String, String> targetHeaders = new HashMap();
+    targetHeaders.put(headerKey, headerVal);
+    when(targetHttpResponse.getHeaders()).thenReturn(Optional.of(targetHeaders));
+
+    // these are needed to make sure the method doesn't Result.FAILURE-out
+    when(targetHttpResponse.getBody()).thenReturn(Optional.of(body));
     when(targetHttpResponse.getCallerResponseOutputStream()).thenReturn(responseStream);
 
     processor.writeTargetResponseOnCaller(targetHttpResponse);
+    verify(listenerHeaders).putAll(targetHeaders);
+    verify(listenerHeaders).remove(headerKey);
+  }
 
-    assertThat(listenerResponse.getHeaders(), not(hasEntry("transfer-encoding", "chunked")));
```

Comment:
this line isn't actually validating that we removed the header because the `listenerResponse` is a mock (and is not configured to return anything validate-able)